plugins/woocommerce/src/Database/Migrations/MigrationHelper.php

```diff
@@ -75,4 +77,163 @@ public static function generate_on_duplicate_statement_clause( array $columns ):
 		return $db_util->generate_on_duplicate_statement_clause( $columns );
 	}
 
+	/**
+	 * Migrate state codes in all the required places in the database (hopefully) when they change for a given country.
+	 *
+	 * @param string $country_code The country that has the states for which the migration is needed.
+	 * @param array  $old_to_new_states_mapping An associative array where keys are the old state codes and values are the new state codes.
+	 * @return void
+	 */
+	public static function migrate_country_states( string $country_code, array $old_to_new_states_mapping ): void {
+		self::migrate_country_states_for_shipping_locations( $country_code, $old_to_new_states_mapping );
+
+		// We'll migrate only the authoritative orders table,
+		// the sync mechanism (if enabled) will take care of updating the backup table.
+		if ( OrderUtil::custom_orders_table_usage_is_enabled() ) {
+			self::migrate_country_states_for_cot_orders( $country_code, $old_to_new_states_mapping );
+		} else {
+			self::migrate_country_states_for_post_orders( $country_code, $old_to_new_states_mapping );
+		}
+
+		self::migrate_country_states_for_store_location( $country_code, $old_to_new_states_mapping );
+	}
+
+	/**
+	 * Migrate state codes in the shipping locations table.
+	 *
+	 * @param string $country_code The country that has the states for which the migration is needed.
+	 * @param array  $old_to_new_states_mapping An associative array where keys are the old state codes and values are the new state codes.
+	 * @return void
+	 */
+	private static function migrate_country_states_for_shipping_locations( string $country_code, array $old_to_new_states_mapping ): void {
+		global $wpdb;
+
+		// phpcs:disable WordPress.DB.PreparedSQL.NotPrepared
+
+		$sql            = "SELECT location_id, location_code FROM {$wpdb->prefix}woocommerce_shipping_zone_locations WHERE location_code LIKE '{$country_code}:%'";
+		$locations_data = $wpdb->get_results( $sql, ARRAY_A );
+
+		foreach ( $locations_data as $location_data ) {
+			$old_state_code = substr( $location_data['location_code'], 3 );
+			if ( array_key_exists( $old_state_code, $old_to_new_states_mapping ) ) {
+				$new_location_code = "{$country_code}:{$old_to_new_states_mapping[$old_state_code]}";
+				$update_query      = $wpdb->prepare(
+					"UPDATE {$wpdb->prefix}woocommerce_shipping_zone_locations SET location_code=%s WHERE location_id=%d",
+					$new_location_code,
+					$location_data['location_id']
+				);
+				$wpdb->query( $update_query );
+			}
+		}
+
+		// phpcs:enable WordPress.DB.PreparedSQL.NotPrepared
```

Comment:
Can this exclusion apply only to the first SELECT?

Reply:
Nope, PHPCS was complaining about it in all the queries of the method.